### PR TITLE
Config: use new treewalker property to skip non-compiled files in regression

### PR DESCRIPTION
**Pull request by mahfouz72**
blocked until https://github.com/checkstyle/checkstyle/pull/15204

### CODE REVIEW

**Comment by mahfouz72** [comment]
@romani ping please can you help to fix CI

`failed during checkstyle configuration: Property 'skipFileOnJavaParseException' does not exist, please check the documentation`. Do we need to get a release first to detect new properties?

**Review by romani — APPROVED** [review]

**Comment by rnveach** [comment]
https://app.circleci.com/pipelines/github/checkstyle/contribution/357/workflows/3b96ba65-6f9f-4f5a-aeaf-0685f220ed0f/jobs/2476
> com.puppycrawl.tools.checkstyle.api.CheckstyleException: Property 'skipFileOnJavaParseException' does not exist, please check the documentation

@mahfouz72 This is not a Checker property.

**Review by rnveach — CHANGES_REQUESTED** [review]
https://github.com/checkstyle/contribution/pull/877#issuecomment-2226758425

**Comment by mahfouz72** [comment]
@rnveach done

File: checkstyle-tester/my_check.xml [hunk 1]
```diff
@@ -27,10 +27,15 @@
...
          <!-- <module name="NestedSwitchCheck"/> -->
 
-         <!-- usuppress javadoc parsing errors, as we test Check not a parser -->
+         <!-- suppress javadoc parsing errors, as we test Check not a parser -->
          <module name="SuppressionXpathSingleFilter">
             <property name="message" value="Javadoc comment at column \d+ has parse error"/>
          </module>
+
+         <!-- as we run on regression even on non-compiled files we need to skip exceptions on them -->
+         <property name="skipFileOnJavaParseException" value="true"/>
```
Comment (romani) [inline checkstyle-tester/my_check.xml:36]: Please put them under other Checker properties. Group them together above , before first inner module

**Review by romani — CHANGES_REQUESTED** [review]
Item

File: checkstyle-tester/my_check.xml [hunk 2]
```diff
@@ -27,10 +27,15 @@
          <!-- Example of sevntu.checkstyle Check usage -->
          <!-- <module name="NestedSwitchCheck"/> -->
 
-         <!-- usuppress javadoc parsing errors, as we test Check not a parser -->
+         <!-- suppress javadoc parsing errors, as we test Check not a parser -->
```
Comment (nrmancuso) [inline checkstyle-tester/my_check.xml:30]: This has been killing me for years, thank you for fixing this.

**Review by nrmancuso — APPROVED** [review]

File: checkstyle-tester/my_check.xml (hunk 1 above)
Comment (mahfouz72) [reply checkstyle-tester/my_check.xml:36]: > Please put them under other Checker properties.

This is not checker property. Do you mean at the first line in tree walker module? like this

```XML
    <module name="TreeWalker">
         <!-- as we run on regression even on non-compiled files we need to skip exceptions on them -->
         <property name="skipFileOnJavaParseException" value="true"/>
         <property name="javaParseExceptionSeverity" value="ignore"/>

         <!-- Example of checkstyle Check usage -->
         <!-- PLEASE CHANGE IT TO CHECK YOU ARE TESTING !!!! -->
         <module name="ThrowsCount">
             <property name="max" value="20000000"/>
         </module>

         <!-- Example of sevntu.checkstyle Check usage -->
         <!-- <module name="NestedSwitchCheck"/> -->

         <!-- suppress javadoc parsing errors, as we test Check not a parser -->
         <module name="SuppressionXpathSingleFilter">
            <property name="message" value="Javadoc comment at column \d+ has parse error"/>
         </module>
    </module>
```

Comment (mahfouz72) [reply checkstyle-tester/my_check.xml:36]: I prefer to put it after the filter and leave the check modules above, right after `<module name="TreeWalker">`

Comment (romani) [reply checkstyle-tester/my_check.xml:36]: Treewalker, yes, all propertties are on most top before any inner modules.

**Comment by romani** [comment]
@relentless-pursuit , please subscribe to this PR, as we merge please update test-configs repo with new content of template config.

File: checkstyle-tester/my_check.xml (hunk 1 above)
Comment (mahfouz72) [reply checkstyle-tester/my_check.xml]: Okay, done

**Review by romani — APPROVED** [review]

**Pull request merged**